src/carousel/carousel.ts

```diff
@@ -280,7 +280,30 @@ export class NgbCarousel implements AfterContentChecked,
       });
     }
 
-    this.slides.changes.pipe(takeUntil(this._destroy$)).subscribe(() => this._cd.markForCheck());
+    this.slides.changes.pipe(takeUntil(this._destroy$)).subscribe(() => {
+      const transitionIds = this._transitionIds;
```

Comment:
nitpick: why have `const transitionIds` ?

Shorter version:
```ts
this._transitionIds?.forEach(id => ngbCompleteTransition(this._getSlideElement(id)));
this._transitionIds = null;
```

Reply:
Yes, better version indeed ! I'm not used enough with the optional chaining operator.